src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,115 @@ private function createRelationsForItem($item, $formattedRelations)
             }
         }
     }
 
+    /**
+     * When using the HasMany/MorphMany relations as selectable elements we use this function to "mimic-sync" in those relations.
+     * Since HasMany/MorphMany does not have the `sync` method, we manually re-create it.
+     * Here we add the entries that developer added and remove the ones that are not in the list.
+     * This removal process happens with the following rules:
+     * - by default Backpack will behave like a `sync` from M-M relations: it deletes previous entries and add only the current ones.
+     * - `force_delete` is configurable in the field, it's `true` by default. When false, if connecting column is nullable instead of deleting the row we set the column to null.
+     * - `fallback_id` could be provided. In this case instead of deleting we set the connecting key to whatever developer gives us.
+     *
+     * @return void
+     */
+    public function attachManyRelation($item, $relation, $relationMethod, $relationDetails, $relation_values)
+    {
+        $model_instance = $relation->getRelated();
+        $force_delete = $relationDetails['force_delete'];
+        $relation_foreign_key = $relation->getForeignKeyName();
+        $relation_local_key = $relation->getLocalKeyName();
+
+        $relation_column_is_nullable = $model_instance->isColumnNullable($relation_foreign_key);
+
+        if ($relation_values !== null) {
+            // we add the new values into the relation
+            $model_instance->whereIn($model_instance->getKeyName(), $relation_values)
+                ->update([$relation_foreign_key => $item->{$relation_local_key}]);
+
+            // we clear up any values that were removed from model relation.
+            // if developer provided a fallback id, we use it
+            // if column is nullable we set it to null if developer didn't specify `force_delete => true`
+            // if none of the above we delete the model from database
+            if ($relationDetails['fallback_id'] !== false) {
+                $model_instance->whereNotIn($model_instance->getKeyName(), $relation_values)
+                    ->where($relation_foreign_key, $item->{$relation_local_key})
+                    ->update([$relation_foreign_key => $relationDetails['fallback_id']]);
+            } else {
+                if (! $relation_column_is_nullable || $force_delete) {
+                    $model_instance->whereNotIn($model_instance->getKeyName(), $relation_values)
+                        ->where($relation_foreign_key, $item->{$relation_local_key})
+                        ->delete();
+                } else {
+                    $model_instance->whereNotIn($model_instance->getKeyName(), $relation_values)
+                        ->where($relation_foreign_key, $item->{$relation_local_key})
+                        ->update([$relation_foreign_key => null]);
+                }
+            }
```

Comment:
```suggestion
            $removed_entries = $model_instance->whereNotIn($model_instance->getKeyName(), $relation_values)->where($relation_foreign_key, $item->{$relation_local_key});
            if ($relationDetails['fallback_id'] !== false) {
                    $removed_entries->update([$relation_foreign_key => $relationDetails['fallback_id']]);
            } else {
                if (! $relation_column_is_nullable || $force_delete) {
                    $removed_entries->delete();
                } else {
                    $removed_entries->update([$relation_foreign_key => null]);
                }
            }
```

Reply:
I see the same query is made 3 times here. We can move it before the conditional to only do it once. Not sure I understood correctly what it does, and chose a good variable name.

Reply:
Done! 👍